lib/screenplay/outfront/sftp.ex

```diff
@@ -11,29 +11,38 @@ defmodule Screenplay.Outfront.SFTP do
   @retries 3
   @sftp_client_module Application.compile_env(:screenplay, :sftp_client_module)
 
-  def set_takeover_images(stations, portrait_png, landscape_png) do
+  def run(work_fn) do
     conn = start_connection()
 
-    Enum.each(stations, fn station ->
-      write_image(conn, station, @portrait_dir, portrait_png)
-      write_image(conn, station, @landscape_dir, landscape_png)
-    end)
+    try do
+      work_fn.(conn)
+    rescue
```

Comment:
I'm wondering about the purpose of this `rescue`. Is there a reason we shouldn't either "let it crash" or return an error to the caller? Generally, it's a good idea to avoid silently swallowing errors (from the caller's perspective, even if the error is logged somewhere as a side effect).

Reply:
The main purpose was to log to Sentry. The big gotcha that I missed is we don't have Sentry logging for server errors at the moment 🙃 . That's not great cause that was what I assumed we would use for getting notified of these issues... I'll remove the `rescue` regardless because if we set up Sentry correctly, error logs will show up there automatically.

Reply:
Good thing we tested it 😅 Will you set that up as part of this task? Seems pretty essential to the "feature".

Reply:
Yeah, I think it's important enough to be in scope for this task. Will get started on that and make sure it's working before merging this branch.

Reply:
Looks like I was fooled by how Screenplay was set up in Sentry. The platform selected in the Project Settings was React so I assumed it was only set up for the frontend. We do have the necessary back configuration for server errors as well. Only thing missing was a separate DSN for dev-green (was still using the dev DSN. PR for that change [here](https://github.com/mbta/devops/pull/1639)). I changed the platform and project name in Sentry so it is set up the way Screens is (which we seem happy with so a good standard for Screenplay).

Reply:
Wait, it needs a separate DSN? I thought the environment name was just configured in the app and sent along with error reports. I don't remember having to set up more than one DSN per project.

Reply:
Very interesting. For whatever reason, I thought that each environment needed their own DSN when I initially set up Sentry for our apps. You're right though, we can use the `environment_name` to identify environment and set up alerts based on that. I closed that devops PR. Not going to mess with existing DSNs for now.